src/python/espressomd/system.pyx

```diff
@@ -330,7 +330,7 @@ cdef class System(object):
             for j in range(_state_size_plus_one + 1):
                 states_on_node_i.append(
                     rng.randint(0, numeric_limits[int].max()))
-            states[i] = " ".join(map(str, states_on_node_i))
+            states[i] = (" ".join(map(str, states_on_node_i))).encode('utf-8')
```

Comment:
Wouldn't it be easier to use a Unicode literal string? (untested)
```python
states[i] = u" ".join(map(str, states_on_node_i))
```

Reply:
It does not work with either of `u` or `b` as a prefix. As we store string in a `std::vector[std::string]` at the Cython level for Python 2/3 compatibility, I did proceed by trial and error and then found the SO answer I mention in the PR.